universal-application-tool-0.0.1/app/services/applicant/ApplicantQuestion.java

```diff
@@ -42,17 +42,7 @@ public String getQuestionText() {
     }
   }
 
-  public ImmutableSet<ValidationErrorMessage> getQuestionErrors() {
```

Comment:
I am thinking to move the switch up here and make all questions to return errors and the `hasErrors` just check on if the set is empty.

Reply:
That sounds good. I think I wrote this originally without thinking about the fact that even the question-level validations would be type-dependent. Your updates make more sense.

Reply:
Ahh, I think I might have misunderstood / misled. We still want to be able to get the "question-level" errors separately from the "field-level" errors (e.g., first name error). This is for rendering purposes. We want to be able render the question-level errors in perhaps a different location / style than the field errors.

With that in mind, maybe we either want to keep it how it was, or add a method on the subclasses like getQuestionErrors.

I'll take a closer look at the overall PR tomorrow tho. :)

Reply:
Oops. Sounds good! Thanks for pointing out. Let me see what I can do here.

Reply:
Done. Let me know what you think.